domains/transfer-agreements/TransferAgreements.API/API.IntegrationTests/Data/TransferAgreementRepositoryTest.cs

```diff
@@ -3,52 +3,45 @@
 using System.Linq;
 using System.Threading.Tasks;
 using API.Data;
+using API.IntegrationTests.Factories;
 using API.IntegrationTests.Testcontainers;
 using API.Models;
 using Microsoft.EntityFrameworkCore;
 using Xunit;
 
 namespace API.IntegrationTests.Data;
 
-public class TransferAgreementRepositoryTest : IClassFixture<PostgresContainer>
+public class TransferAgreementRepositoryTest : IClassFixture<TransferAgreementsApiWebApplicationFactory>
 {
-    private readonly DbContextOptions<ApplicationDbContext> options;
+    private readonly TransferAgreementsApiWebApplicationFactory factory;
 
-    public TransferAgreementRepositoryTest(PostgresContainer dbContainer)
+    public TransferAgreementRepositoryTest(TransferAgreementsApiWebApplicationFactory factory)
     {
-        options = new DbContextOptionsBuilder<ApplicationDbContext>().UseNpgsql(dbContainer.ConnectionString).Options;
+        this.factory = factory;
     }
 
     [Fact]
     public async Task AddTransferAgreementToDb_SameSenderIdAndTransferAgreementNumber_ShouldThrowException()
     {
-        await using var dbContext = new ApplicationDbContext(options);
-        await dbContext.Database.EnsureCreatedAsync();
-        var repository = new TransferAgreementRepository(dbContext);
-
         var senderId = Guid.NewGuid();
         var agreements = SetupTransferAgreements(senderId);
+        await factory.SeedTransferAgreements(agreements);
 
-        foreach (var agreement in agreements)
-        {
-            await repository.AddTransferAgreementToDb(agreement);
-        }
-
-        var existingTransferAgreement = dbContext.TransferAgreements.First();
 
         var transferAgreement = new TransferAgreement()
         {
+            Id = Guid.NewGuid(),
             StartDate = DateTimeOffset.UtcNow,
             EndDate = DateTimeOffset.UtcNow.AddDays(10),
             SenderId = senderId,
             SenderName = "nrgi A/S",
             SenderTin = "44332211",
             ReceiverTin = "12345678",
             ReceiverReference = Guid.NewGuid(),
-            TransferAgreementNumber = existingTransferAgreement.TransferAgreementNumber
+            TransferAgreementNumber = agreements[0].TransferAgreementNumber
         };
-        dbContext.TransferAgreements.Add(transferAgreement);
-        await Assert.ThrowsAsync<DbUpdateException>(() => dbContext.SaveChangesAsync());
+
+        await Assert.ThrowsAsync<DbUpdateException>(() => factory.SeedTransferAgreementsSaveChangesAsync(transferAgreement));
```

Comment:
I see that the way you are adding transfer agreements to the db, is directly through seeding the database. Is that the intended approach, or would it be closer to real-life use case, if you tried adding the transferagreements through the API?